packages/vite/src/node/build.ts

```diff
@@ -236,6 +236,13 @@ export interface BuildOptions {
    * @default null
    */
   watch?: WatcherOptions | null
+  /**
+   * Watch config files and restart the server when they changed.
+   * When enabled, the build function will return `() => RollupWatcher`
+   * This will be set to true by default and be removed in Vite 5.
+   * @default false
+   */
+  watchConfig?: boolean
```

Comment:
I was thinking we add this param to `build()` directly since it only needs to be controlled programatically.